libr/asm/arch/pyc/pyc_dis.c

```diff
@@ -0,0 +1,217 @@
+/* radare - LGPL3 - Copyright 2016-2020 - c0riolis, x0urc3 */
+
+#include "pyc_dis.h"
+
+static const char *cmp_op[] = { "<", "<=", "==", "!=", ">", ">=", "in", "not in", "is", "is not", "exception match", "BAD" };
+
+int r_pyc_disasm (RAsmOp *opstruct, const ut8 *code, RList *cobjs, RList *interned_table, ut64 pc, pyc_opcodes *ops) {
+	pyc_code_object *cobj = NULL, *t = NULL;
+	ut32 extended_arg = 0, i = 0, oparg;
+	st64 start_offset, end_offset;
+	RListIter *iter = NULL;
+
+	char *name = NULL;
+	const char *arg = NULL;
+	RList *varnames;
+	RList *consts;
+	RList *names;
+	RList *freevars;
+	RList *cellvars;
+	ut8 op;
+
+	r_list_foreach (cobjs, iter, t) {
+		start_offset = t->start_offset;
+		end_offset = t->end_offset;
+		if (start_offset <= pc && pc < end_offset) { // pc in [start_offset, end_offset)
+			cobj = t;
+			break;
+		}
+	}
+
+	if (cobj != NULL) {
+		/* TODO: adding line number and offset */
+		varnames = cobj->varnames->data;
+		consts = cobj->consts->data;
+		names = cobj->names->data;
+		freevars = cobj->freevars->data;
+		cellvars = cobj->cellvars->data;
+
+		op = code[i];
+		i += 1;
+		name = ops->opcodes[op].op_name;
+		r_strbuf_set (&opstruct->buf_asm, name);
+		if (name == NULL) {
+			return 0;
+		}
+		if (op >= ops->have_argument) {
+			if (ops->bits == 16) {
+				oparg = code[i] + code[i + 1] * 256 + extended_arg;
+				i += 2;
+			} else {
+				oparg = code[i] + extended_arg;
+				i += 1;
+			}
+			extended_arg = 0;
+			if (op == ops->extended_arg) {
+				if (ops->bits == 16) {
+					extended_arg = oparg * 65536;
+				} else {
+					extended_arg = oparg << 8;
+				}
+			}
+			arg = parse_arg (&ops->opcodes[op], oparg, names, consts, varnames, interned_table, freevars, cellvars, ops->opcode_arg_fmt);
+			if (arg != NULL) {
+				r_strbuf_appendf (&opstruct->buf_asm, "%20s", arg);
+			}
+		} else if (ops->bits == 8) {
+			i += 1;
+		}
+
+		return i;
+	}
+	return 0;
+}
+
+const char *parse_arg (pyc_opcode_object *op, ut32 oparg, RList *names, RList *consts, RList *varnames, RList *interned_table, RList *freevars, RList *cellvars, RList *opcode_arg_fmt) {
+	pyc_object *t = NULL;
+	const char *arg = NULL;
+	pyc_code_object *tmp_cobj;
+	pyc_arg_fmt *fmt;
+	RListIter *i = NULL;
+
+	// version-specific formatter for certain opcodes
+	r_list_foreach (opcode_arg_fmt, i, fmt)
+		if (!strcmp (fmt->op_name, op->op_name)) {
+			return fmt->formatter (oparg);
+		}
+
+	if (op->type & HASCONST) {
+		t = (pyc_object *)r_list_get_n (consts, oparg);
```

Comment:
Define and assign in the same line

Reply:
No, `t` will be used in more than one scope with same meaning.